src/pybind/mgr/dashboard/controllers/oauth2.py

```diff
@@ -0,0 +1,29 @@
+import cherrypy
+
+from dashboard.controllers._auth import ControllerAuthMixin
+from dashboard.exceptions import DashboardException
+from dashboard.services.auth.oauth2 import OAuth2
+
+from .. import mgr
+from . import BaseController, Endpoint, Router
+
+
+@Router('/auth/oauth2', secure=False)
+class Oauth2(BaseController, ControllerAuthMixin):
+
+    @Endpoint(json_response=False, version=None)
+    def login(self):
+        if not mgr.get_module_option('sso_oauth2'):
+            raise DashboardException(500, msg='Failed to login: SSO OAuth2 is not enabled')
```

Comment:
This is still exposing implementation details from the PoV of a controller. The only thing that matters to this controller is whether OAuth2 is enabled or not, but not what you have to do to find that out:
```suggestion
        if not OAuth2.enabled():
            raise DashboardException(500, msg='Failed to login: SSO OAuth2 is not enabled')
```

Reply:
Thanks, got it! Updated it